src/main/java/uk/gov/hmcts/ccd/domain/service/processor/DateTimeFormatParser.java

```diff
@@ -49,11 +39,38 @@ public String convertIso8601ToDateTime(String dateTimeFormat, String value) {
     }
 
     public String convertIso8601ToDate(String dateFormat, String value) {
-        LocalDate date = LocalDate.parse(value, DATE_FORMAT);
+        LocalDate date = LocalDate.parse(value, DateTimeFormatter.ofPattern(DATE_FORMAT));
         return date.format(DateTimeFormatter.ofPattern(dateFormat));
     }
 
-    private DateTimeFormatter getDateTimeFormatter(String dateTimeFormat) {
+    private String convert(List<String> permittedInputFormats, String outputFormat, String value) {
+        List<DateTimeFormatter> inputFormatters = permittedInputFormats.stream()
+            .filter(Objects::nonNull)
+            .map(this::getFormatter)
+            .collect(Collectors.toList());
+        DateTimeFormatter outputFormatter = getFormatter(outputFormat);
+        TemporalAccessor parsed = null;
+
+        Iterator<DateTimeFormatter> iterator = inputFormatters.iterator();
+        while (iterator.hasNext()) {
+            DateTimeFormatter inputFormatter = iterator.next();
+
+            ParsePosition parsePosition = new ParsePosition(0);
+            parsed = inputFormatter.parseUnresolved(value, parsePosition);
+
+            if (parsed != null && isValidParseResult(value, parsePosition)) {
+                break;
+            } else if (!iterator.hasNext()) {
+                throw new DateTimeParseException(
+                    String.format("Failed to parse value '%s' against formats: %s",
```

Comment:
This will add **null** as well into the exception message as permittedInputFormats may contain **null** values